webapps/frontend/app/viewmodels/wizards/createfromtemplatewizard/gather.js

```diff
@@ -155,6 +178,13 @@ define([
                 self.data.vm().load();
                 self.data.selectedPMachines([]);
             }
+            generic.xhrAbort(self.loadVMachinesHandle);
+            self.loadVMachinesHandle = api.get('vmachines/', { queryparams: {contents: 'name'}})
+                .done(function(data) {
+                    $.each(data.data, function(index, item) {
+                        self.data.vMachinesNames.push(item.name);
```

Comment:
I'd expect that somewhere ```self.data.vMachineNames``` would be reset to an empty list (typically in index.js), because now this list will grow larger with every time the wizard is opened.